pkg/aws/commandbuilder.go

```diff
@@ -0,0 +1,126 @@
+	DELETE_ROLE_POLICY               Command = "delete-role-policy"
+	ATTACH_ROLE_POLICY               Command = "attach-role-policy"
+	DETACH_ROLE_POLICY               Command = "detach-role-policy"
+	TAG_POLICY                       Command = "tag-policy"
+	TAG_ROLE                         Command = "tag-role"
+	CREATE_OPEN_ID_CONNECT_PROVIDER  Command = "create-open-id-connect-provider"
+	DELETE_OPEN_ID_CONNECT_PROVIDER  Command = "delete-open-id-connect-provider"
+	DELETE_ROLE_PERMISSIONS_BOUNDARY Command = "delete-role-permissions-boundary"
+)
+
+type Param string
+
+const (
+	UNDEFINED_PARAM              Param = "undefined param"
+	TAGS                         Param = "tags"
+	ROLE_NAME                    Param = "role-name"
+	ASSUME_ROLE_POLICY_DOCUMENT  Param = "assume-role-policy-document"
+	PERMISSIONS_BOUNDARY         Param = "permissions-boundary"
+	PATH                         Param = "path"
+	POLICY_NAME                  Param = "policy-name"
+	POLICY_DOCUMENT              Param = "policy-document"
+	POLICY_ARN                   Param = "policy-arn"
+	URL                          Param = "url"
+	CLIENT_ID_LIST               Param = "client-id-list"
+	THUMBPRINT_LIST              Param = "thumbprint-list"
+	OPEN_ID_CONNECT_PROVIDER_ARN Param = "open-id-connect-provider-arn"
+	SET_AS_DEFAULT               Param = "set-as-default"
+)
+
+type ICommandBuilder interface {
+	SetService(awsService Service)
+	SetCommand(awsCommand Command)
+	AddParam(awsParam Param, value interface{})
+	ResetParams()
+	Build() string
+}
+
+type CommandBuilder struct {
+	service string
+	command string
+	params  []string
+}
+
+type StringMap map[string]string
+
+func (b *CommandBuilder) SetService(awsService Service) {
+	b.service = fmt.Sprint(awsService)
+}
+
+func (b *CommandBuilder) SetCommand(awsCommand Command) {
+	b.command = fmt.Sprint(awsCommand)
+}
+
+func (b *CommandBuilder) AddParam(awsParam Param, value interface{}) {
+	if b.params == nil {
+		b.params = make([]string, 1)
+	}
+	stringValue := ""
+	typeOf := fmt.Sprintf("%T", value)
+	switch typeOf {
+	case "string":
+		stringValue = value.(string)
+	case "map[string]string":
+		stringValue = createKeyValuePairs(value.(map[string]string))
+	default:
+		stringValue = ""
+	}
+
+	if stringValue != "" {
+		b.params = append(b.params, fmt.Sprintf("\t--%s %s", awsParam, stringValue))
+	}
+}
+
+func (b *CommandBuilder) AddBoolParam(awsParam Param) {
+	if b.params == nil {
+		b.params = make([]string, 1)
+	}
+	b.params = append(b.params, fmt.Sprintf("\t--%s", awsParam))
+}
+
+func (b *CommandBuilder) ResetParams() {
+	b.params = nil
+}
+
+func (b *CommandBuilder) Build() string {
+	return fmt.Sprintf("aws %s %s%s", b.service, b.command, strings.Join(b.params, " \\\n"))
+}
+
+func createKeyValuePairs(m map[string]string) string {
+	b := new(bytes.Buffer)
+	keys := make([]string, 0, len(m))
+	for k := range m {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
```

Comment:
Not really, but since order does not matter it changes depending on the execution. Ordering makes sure we always get the same output, so we can test it later more easily